daemon/logger/loggerutils/rotatefilewriter.go

```diff
@@ -100,17 +100,6 @@ func rotate(name string, maxFiles int) error {
 
 // backup renames a file from fromPath to toPath
 func backup(fromPath, toPath string) error {
-	if _, err := os.Stat(fromPath); os.IsNotExist(err) {
-		return err
-	}
-
-	if _, err := os.Stat(toPath); !os.IsNotExist(err) {
-		err := os.Remove(toPath)
-		if err != nil {
-			return err
-		}
-	}
-
 	return os.Rename(fromPath, toPath)
```

Comment:
To keep the same logic for the caller you need to catch the LinkError here and see if it is an `ENOENT` then return an `os.ErrNotExist` to the caller